php/libraries/User.class.inc

```diff
@@ -541,5 +541,24 @@ class User extends UserPermissions
         return $this->hasPermission($code)
             && $this->hasCenter($center_id);
     }
+
+    /**
+     * Determines if the user has ever logged in successfully
+     *
+     * @return bool
+     */
+    public function hasEverLoggedIn(): bool
```

Comment:
what if the user has a password hash (which I would assume means a valid account) but hasn't used the password to log in yet?

Reply:
It would return `false`. This function only check if the user has ever logged in.

Reply:
right..should we still in that case be able to reject the account just because they have not yet logged in but still intend to as implied by the existence of a password hash, which in turn implies a valid account? i would think not. This comment really refers to line :831

Reply:
ohhh, can you only get a password hash after you've logged in?